src/components/FriendItem/FriendItem.js

```diff
@@ -1,19 +1,38 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import { Button } from 'reactstrap';
 import friendShape from '../../helpers/props/friendShape';
 import './FriendItem.scss';
 
 class FriendItem extends React.Component {
   static propTypes = {
     friend: friendShape,
+    status: PropTypes.string,
+    endFriendship: PropTypes.func,
   };
 
+  deleteEvent = (e) => {
+    e.preventDefault();
+    const friendRequestId = e.target.id;
```

Comment:
change this to:
`const friendRequestId = e.target.closest('button').id`
to avoid the issue you are getting with sometimes deleting everything